### DISCUSSION TITLE
Auto-engage gear using gear signal?

### DISCUSSION THREAD

**Discussion by zeynelozturk** [danielnilsson9/bbs-fw#121] [Ideas]
Would it be a good idea if the gears would "auto-engage" using gear sensor? It would be great when just using the throttle. Then when freewheeling, you could slowly change multiple gears without touching the throttle at all.

It would work similar to;

If (gearSensorSignal = true && 
wheel is spinning above x rpm && currentPedalCadence < someValue &&
CurrentAmps < 0.5)
{
GiveMotor10WattsForASecToEngage()
//Above can limit the motor rpm too, to change gear smoothly.
}

This should definitely not spin the motor if the bike is stopped. I don't know if there would be serious disadvantages to a such feature.

**Comment by EigenMania** [comment]
I proposed something to this effect a while back. My idea was to have a constant (very low) pre-tension current to keep slack out of the chain, which would kick in only above a certain roadspeed (so independent of pedal cadence and gear sensor). 

For me the primary goal was to keep slack out of the chain so that when throttle is applied the freewheel pawls do not get hit hard. But it could also work to auto-engage gears when coasting. My bike has a hub gear so I could not test this precisely, but you could if motivated :)

The branch is still available (for me it's a useful feature), and up-to-date with master, so feel free to try it out: https://github.com/EigenMania/bbs-fw/tree/feature/apply_pretension

**Reply by zeynelozturk** [reply]
Yes, I read it and I think pre-tensioning would even make my feature suggestion kind of unnecessary. But it's also constant power usage.

When I think about my suggestion, I think such feature will need a good test and/or some config parameters. Because gear sensor will send a signal when you started shifting and if auto-engage power is not low enough, it'll probably do some crunching.

**Reply by kb79000** [reply]
> J'ai proposé quelque chose à cet effet il y a quelque temps. Mon idée était d'avoir un courant de pré-tension constant (très faible) pour maintenir le mou de la chaîne, qui n'interviendrait qu'au-dessus d'une certaine vitesse (donc indépendante de la cadence de pédale et du capteur de vitesse).
> 
> Pour moi, l'objectif principal était de garder le mou de la chaîne afin que lorsque l'accélérateur est appliqué, les cliquets de roue libre ne soient pas durement touchés. Mais cela pourrait également fonctionner pour engager automatiquement les vitesses en roue libre. Mon vélo a un moyeu donc je n'ai pas pu le tester précisément, mais vous pourriez le faire si vous êtes motivé :)
> 
> La branche est toujours disponible (pour moi, c'est une fonctionnalité utile) et à jour avec master, alors n'hésitez pas à l'essayer : https://github.com/EigenMania/bbs-fw/tree/feature/apply_pretension

Hello do you have good result with your new option ?🙂
I would like to try this if you can share the firmware for BBS02B? Thx 😉

**Reply by EigenMania** [reply]
For BBS02B (just delete .txt extensions):

[bbs-fw.hex.txt](https://github.com/danielnilsson9/bbs-fw/files/11732738/bbs-fw.hex.txt)
[BBSFWTool.exe.txt](https://github.com/danielnilsson9/bbs-fw/files/11732741/BBSFWTool.exe.txt)

**Reply by kb79000** [reply]
> For BBS02B (just delete .txt extensions):
> 
> [bbs-fw.hex.txt](https://github.com/danielnilsson9/bbs-fw/files/11732738/bbs-fw.hex.txt) [BBSFWTool.exe.txt](https://github.com/danielnilsson9/bbs-fw/files/11732741/BBSFWTool.exe.txt)

Thank you I will try in this week the speeds pass by lowering the speed and braking at the same time ? Thank  🙂

**Comment by CitizenDev** [comment]
As a gear sensor user I also see this as useful feature. For me it would eliminate the issue of downshifting when freewheeling to a near-stop situation (crossing, stop-sign etc). I quite often find myself in this awkward situation where i'm slightly tapping breaks and freewheeling to slow down but also need to pedal a bit for gear changes to go trough. 

Also the point made by @EigenMania  about pretension lowering force applied to freewheel is absolutely valid since I've broken one set of freewheel by applying too much torque while chain was not tensioned.

**Comment by dav0000000** [comment]
Yes this does sound like a useful option to have. I might play around with it a bit.

**Comment by zeynelozturk** [comment]
Something very similar to this introduced as "FreeShift" by Shimano. They basically say that it's "pedal-free" shifting that helps you when you're descending. 

 There is also Auto shift but it's probably a bit out of topic for this thread as it involves derailleur.

https://bike.shimano.com/en-EU/technologies/component/details/steps/free-shift.html

**Reply by danielnilsson9** [reply]
Sounds like a useful feature, if anyone want to add this and test it, feel free to do so and send a PR and we can work towards having it included.

**Comment by kb79000** [comment]
Hi are there any feedback from users of this function? I would like to test and also bring a return but impossible to compile the firmware without knowledge.
Thx 🙂

**Reply by WvBijsterveld** [reply]
Thinking about finally installing my gear sensor and then i will use the software of EigenMania, will report back later.
